include/triton/core/tritonserver.h

```diff
@@ -362,6 +363,35 @@ typedef TRITONSERVER_Error* (*TRITONSERVER_ResponseAllocatorAllocFn_t)(
     TRITONSERVER_MemoryType* actual_memory_type,
     int64_t* actual_memory_type_id);
 
+/// Type for allocation function that allocates a buffer to hold an
+/// output tensor with buffer attributes.
+///
+/// \param allocator The allocator that is provided in the call to
+/// TRITONSERVER_InferenceRequestSetResponseCallback.
+/// \param tensor_name The name of the output tensor to allocate for.
+/// \param userp The user data pointer that is provided as
+/// 'response_allocator_userp' in the call to
+/// TRITONSERVER_InferenceRequestSetResponseCallback.
+/// \param preferred_buffer_attributes The buffer attributes that the caller
+/// prefers for the buffer.
+/// \param actual_buffer_attributes Returns the buffer attributes associated
+/// with the allocated buffer.
```

Comment:
The response allocator accepts a buffer attributes for both preferred and actual arguments.